packages/core/src/__tests__/acceptance/service.acceptance.ts

```diff
@@ -0,0 +1,203 @@
+  });
+
+  it('injects a service instance matching a sub class', async () => {
+    class MyController {
+      constructor(@service(MyService) public myService: MyService) {}
+    }
+    ctx.unbind('services.MyService');
+    ctx
+      .bind('services.MySubService')
+      .toClass(MySubService)
+      .apply(asService(MyService));
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    const controller = await ctx.get<MyController>('controllers.MyController');
+    expect(controller.myService).to.be.instanceOf(MySubService);
+  });
+
+  it('allows optional flag', async () => {
+    class MyController {
+      constructor(
+        @service(MyService, {optional: true}) public myService?: MyService,
+      ) {}
+    }
+
+    ctx.unbind('services.MyService');
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    const controller = await ctx.get<MyController>('controllers.MyController');
+    expect(controller.myService).to.be.undefined();
+  });
+
+  it('allows serviceInterface as a string', async () => {
+    class MyController {
+      constructor(@service('MyService') public myService: MyService) {}
+    }
+
+    myServiceBinding.tag({[CoreTags.SERVICE_INTERFACE]: 'MyService'});
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    const controller = await ctx.get<MyController>('controllers.MyController');
+    expect(controller.myService).to.be.instanceOf(MyService);
+  });
+
+  it('reports error if serviceInterface not found - string', async () => {
+    class MyController {
+      constructor(@service('MyService') public myService: MyService) {}
+    }
+
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    return expect(
+      ctx.get<MyController>('controllers.MyController'),
+    ).to.be.rejectedWith(/No binding found for MyService/);
+  });
+
+  it('allows serviceInterface as a symbol', async () => {
+    const MyServiceInterface = Symbol('MyService');
+    class MyController {
+      constructor(@service(MyServiceInterface) public myService: MyService) {}
+    }
+
+    myServiceBinding.tag({[CoreTags.SERVICE_INTERFACE]: MyServiceInterface});
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    const controller = await ctx.get<MyController>('controllers.MyController');
+    expect(controller.myService).to.be.instanceOf(MyService);
+  });
+
+  it('reports error if serviceInterface not found - symbol', async () => {
+    const MyServiceInterface = Symbol('MyService');
+    class MyController {
+      constructor(@service(MyServiceInterface) public myService: MyService) {}
+    }
+
+    ctx.bind('controllers.MyController').toClass(MyController);
+
+    return expect(
+      ctx.get<MyController>('controllers.MyController'),
+    ).to.be.rejectedWith(/No binding found for Symbol\(MyService\)/);
+  });
+
+  it('allows skipSubClasses flag', async () => {
+    class MyController {
+      constructor(
+        @service(MyService, {skipSubClasses: true})
```

Comment:
AFAICT, the flag `skipSubClasses` is no longer implemented. I think this test is no longer relevant. Can you remove it?